qdl.c

```diff
@@ -391,6 +391,9 @@ static void print_usage(void)
 	fprintf(stderr,
 		"%s [--debug] [--storage <emmc|nand|ufs>] [--finalize-provisioning] [--include <PATH>] <prog.mbn> [<program> <patch> ...]\n",
 		__progname);
+	fprintf(stderr,
+		"%s --ramdump [segment-filter,...] [--ramdump-path <PATH>]\n",
+		__progname);
```

Comment:
I think, this looks like a completely separate tool from the UX point of view. You can not mix download and rumdump at the same time, can you?

Reply:
Correct, you would either download or ramdump at any point in time, as such I did consider implementing it as a separate binary. But it's two modes of a very similar thing, so it felt reasonable to associate them both with the same command, and I didn't want to clutter people's /usr/local/bin unnecessarily.

Reply:
I'd still suggest two commands. It sounds very strange to call the same binary in this way. But maybe it's just me :-D

Reply:
I associate the two states/operations and hence find it convenient to only have to have one tool laying around for mucking with that (and only one tool to remember the name of). I'd expect/hope distros pack both binaries, so that the user have them both...

What do you want the ramdump binary to be called then?

Reply:
qramdump